packages/java-edition/src/mcfunction/node/argument.ts

```diff
@@ -304,6 +349,61 @@ export namespace ItemNode {
 	}
 }
 
+export interface ComponentListNode extends core.AstNode {
+	type: 'mcfunction:component_list'
+	children: core.PairNode<core.ResourceLocationNode, nbt.NbtNode>[]
+}
+
+export namespace ComponentListNode {
+	export function is(node: core.AstNode): node is ComponentListNode {
+		return (node as ComponentListNode).type === 'mcfunction:component_list'
+	}
+}
+
+export interface ComponentPredicatesNode extends core.AstNode {
+	type: 'mcfunction:component_predicates'
+	children: ComponentTestBaseNode[]
```

Comment:
Instead of `ComponentTestBaseNode` I think you'd want to create a new union type `type ComponentTestNode = ComponentTestExactNode | ComponentTestExistsNode | ComponentTestSubpredicateNode` and use that here as the children.